dietpi/dietpi-software

```diff
@@ -15257,6 +15295,15 @@ _EOF_
 			fi
 
 		fi
+		
+		software_id=134 # Docker Compose
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == -1 )); then
+
+			Banner_Uninstalling
+			
+			G_EXEC_OUTPUT=1 G_EXEC pip3 uninstall -y docker-compose
```

Comment:
Shall we allow this failing, in case pip got removed already?
```
G_EXEC_NOHALT=1 G_EXEC_OUTPUT=1 G_EXEC pip3 uninstall -y docker-compose
```
It will still print a red failure but will skip the error prompt and continue. I think for uninstalls it's fine when something fails, or the other way round, it is annoying if an uninstall failed because something has already been uninstalled manually or files gone lost or such, what you think?

Reply:
well hard to say, if we would allow it failing but continue uninstall, we will end up with `docker-compose` still being installed but we would have marked it as uninstalled on `/boot/dietpi/.installed`. Isn't it?

Reply:
Yes exactly. Currently most uninstalls are not error-handled and I didn't actively add strict error-handling since I'm regularly annoyed by e.g. DEB package uninstall scripts that fail if certain files of the package are already missing. In such case all further APT progress is blocked, so it's much worse than it would be in our case. I remember a nightmare with some python packages, where it was nearly impossible to perform that uninstall since malformed files in some of them broke their uninstall which then blocked the other uninstall with broken dependencies error and so on. And reinstall didn't work as well because of broken dependencies. I think 90% of users would have needed to install Debian from scratch, which is insane when it is only about the ability to remove all package files so that APT/DPkg are able to attempt a reinstall or clean up dependencies. So while it's good to aim a clean uninstall, it should be at least possible to continue if something fails, as worst thing is a few left files that usually do not hurt or could/need to be be dealt with manually anyway.

Reply:
ok than let's go with it this way